### PR TITLE
feat(SPV-1112): change models of Transaction and BUMP in BEEF hex creation to go-SDK Transaction and MerklePath

### PR DESCRIPTION
**Pull request by jakubmkowalski**
**DO NOT MERGE UNTIL GO-SDK AND GO-PAYMAIL ARE RELEASED WITH FIXES**
[PR in go-SDK](https://github.com/bitcoin-sv/go-sdk/pull/59)
Merged [PR in go-Paymail](https://github.com/bitcoin-sv/go-paymail/pull/132)
This pull request includes significant changes to the `engine` package. The most important changes include adding a new conversion function, refactoring the BEEF preparation process, and simplifying the transaction collection initialization.

### Refactoring BEEF Preparation:
* [`engine/beef_bump.go`](diffhunk://#diff-f7e181e846d5042ed885ea85b2b156e0107948c15b17d6b055b8d5d619c01d2aL5-R121): Refactored the BEEF preparation process by removing the `calculateMergedBUMP` and `validateBumps` functions and introducing the `beefPreparationContext` struct and related methods to streamline the process.

### Simplifying Transaction Collection Initialization:
* [`engine/beef_bump.go`](diffhunk://#diff-f7e181e846d5042ed885ea85b2b156e0107948c15b17d6b055b8d5d619c01d2aL146-R158): Simplified the `initializeRequiredTxsCollection` function to return only the processed SDK transaction without additional collections.

### Conversion Function Addition:
* [`conv/convert_primitives.go`](diffhunk://#diff-510d50506058c7261b3c150e7a3bb54c8ef2edc688c86d21407e72b806edc197R18-R25): Added a new function `Uint64ToUint32` to convert `uint64` to `uint32` with range checks.

# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: engine/beef_bump.go [hunk 1]
```diff
@@ -2,130 +2,123 @@ package engine
...
-		if bump == nil {
-			continue
+	processedSDKTx := txsForBEEF[0]
+	txID := *processedSDKTx.TxID()
+	bctx.transactionMap[txID] = processedSDKTx
+	bctx.queue = append(bctx.queue, processedSDKTx)
+
+	for len(bctx.queue) > 0 {
+		currentTx := bctx.queue[0]
+		bctx.queue = bctx.queue[1:]
```
Comment (dzolt-4chain) [inline engine/beef_bump.go:42]: Can there be a situation when queue has only 1 item inside? Then this line would error

**Review by wregulski — APPROVED** [review]

File: engine/beef_bump.go (hunk 1 above)
Comment (jakubmkowalski) [reply engine/beef_bump.go:42]: This line will not error when in queue we have only one item.
[Pop Front/Shift](https://go.dev/wiki/SliceTricks#pop-frontshift)
`x, a = a[0], a[1:]`
as long as shift value will be lower the slice capacity there are no issues.
So this line:
`len(bctx.queue) > 0 {`
ensures that capacity will be bigger than 0 when we enter loop

File: engine/beef_bump.go [hunk 2]
```diff
@@ -2,130 +2,123 @@ package engine
...
-			continue
-		}
+type beefPreparationContext struct {
+	ctx            context.Context
+	store          TransactionGetter
+	transactionMap map[chainhash.Hash]*trx.Transaction
+	visited        map[chainhash.Hash]bool
+	queue          []*trx.Transaction
+	txsForBEEF     []*trx.Transaction
+}
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply engine/beef_bump.go]: renamed to `visitedTransactions`

File: engine/beef_bump.go [hunk 3]
```diff
@@ -2,130 +2,123 @@ package engine
...
 
-		bumps[tx.BlockHeight] = append(bumps[tx.BlockHeight], tx.BUMP)
+func prepareBEEFFactors(ctx context.Context, tx *Transaction, store TransactionGetter) ([]*trx.Transaction, error) {
+	txsForBEEF, err := initializeRequiredTxsCollection(tx)
+	if err != nil {
+		return nil, err
 	}
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply engine/beef_bump.go]: logic of `initializeRequiredTxsCollection(tx)` moved to `prepareBEEFFactors`

File: engine/beef_bump.go [hunk 4]
```diff
@@ -2,130 +2,123 @@ package engine
...
-	for _, tx := range txs {
-		if tx.BUMP.BlockHeight == 0 || len(tx.BUMP.Path) == 0 {
-			continue
-		}
+type beefPreparationContext struct {
+	ctx            context.Context
+	store          TransactionGetter
+	transactionMap map[chainhash.Hash]*trx.Transaction
+	visited        map[chainhash.Hash]bool
+	queue          []*trx.Transaction
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply engine/beef_bump.go]: If it is only suggestion I will stay with my implementation as i think that here the queue operations are straightforward and are only used within this function. Introducing a separate queue type might add unnecessary complexity.

File: engine/beef_bump.go [hunk 5]
```diff
@@ -2,130 +2,123 @@ package engine
...
-		return nil, nil, err
-	}
-
-	txIDs := make([]string, 0, len(tx.draftTransaction.Configuration.Inputs))
-	for _, input := range tx.draftTransaction.Configuration.Inputs {
-		txIDs = append(txIDs, input.UtxoPointer.TransactionID)
-	}
-
-	inputTxs, err := getRequiredTransactions(ctx, txIDs, store)
+func (bctx *beefPreparationContext) retrieveAndProcessInputTransactions(inputTxIDs []string, currentTx *trx.Transaction) error {
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply engine/beef_bump.go]: name changed to `processInputTransactions`

File: engine/beef_bump.go (hunk 1 above)
Comment (dzolt-4chain) [reply engine/beef_bump.go:43]: Ahh I see smart :D

**Review by mgosek-4chain — APPROVED** [review]

**Review by dzolt-4chain — APPROVED** [review]

**Pull request merged**